.github/workflows/main_workflow.yml

```diff
@@ -402,6 +402,7 @@ jobs:
           docker pull valory/autonolas-registries:latest
           docker pull valory/contracts-amm:latest
           docker pull valory/safe-contract-net:latest
+          docker pull valory/slow-tendermint-server:0.1.0
```

Comment:
This new image is required to showcase the problem.

Reply:
We cannot use a unit test as the actual error is thrown on the tendermint side. 
This new image simply "lags" in the hard reset response to the agent, leading to the scenario laid out in the description of the PR.

Reply:
Ok I see - so it is not used in production

Reply:
Correct, only used for this specific test. I have added disclaimers in the docstring to not use this anywhere else, as it might lead to unwanted behaviour.